protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,49 @@
+# Automated Market Maker Framework
+
+## Summary
+
+The automated market maker (AMM) framework is designed to allow for the provision of an on-chain market making methodology which automatically provides prices according to a simple set of rules based on current market data. These rulesets are not created with the expectation of providing any profit nor of remaining solvent under any specific conditions, and so should be limited to conceptually simple setups. The initial methodology follows a concentrated-liquidity style constant-function market setup, with configurable maximum and minimum price bounds.
+
+An automated market maker is configured at a per-key level, and is enabled by submitting a transaction with the requisite parameters. Once enabled, the configuration will be queried once per block and the resultant orders will be placed on the book, combined with all current orders from the key being cancelled. The party configuring the market maker is responsible for ensuring that the orders generated meet the requirements for any liquidity commitment which is also associated to the key.
+
+Each party may have only one AMM configuration per market.
+
+## Process Overview
+
+The configuration and resultant lifecycle of an automated market maker is as follows:
+
+- Party funds a key which will be used by the strategy with desired token amounts.
+- Party submits a transaction containing configuration for the strategy on a given market. This may vary, but will likely contain parameters such as:
+  - Maximum position notional/absolute value
+  - Price bounds
+  - Granularity of price levels to post
+- Once accepted, the network will begin to monitor the configured strategy on that key.
+- At each block, immediately prior to the evaluation of liquidity provision for SLA purposes, for each configured AMM:
+  - The party's available balance (including margin and general accounts) for trading on the market will be checked. If the total balance is `0` the AMM configuration will be cancelled. 
+  - Each running AMM will be queried for it's provided orders. All orders from this party on this market will then be cancelled, followed by the placement of these new orders on the book. There are a couple of things to note here:
+  - All orders are placed as-if arriving from the party's key externally.
+  - No bond is taken for the market maker, the party is responsible for ensuring the key remains sufficiently collateralised.
+  - No lock is placed on the key, the party may continue to sign and submit other trading transactions and orders. However, any orders resting on the book will be cancelled alongside others each time the AMM updates.
```

Comment:
Is this necessary? It may be for efficiency but would be nice to avoid.

Reply:
To be honest I quite like the idea of pushing "this key is currently in automated trading mode, don't mess with it" as much as possible :D. But yes I'm mainly thinking from an efficiency point of view such that the core doesn't have to track which orders came from the market maker constantly.

Reply:
We can ask core when it comes to this detail.